paimon-trino-common/src/main/java/org/apache/paimon/trino/TrinoMetadata.java

```diff
@@ -516,4 +559,83 @@ public Optional<LimitApplicationResult<ConnectorTableHandle>> applyLimit(
 
         return Optional.of(new LimitApplicationResult<>(table, false, false));
     }
+
+    private static Map<TrinoColumnHandle, Domain> handleElementAtArguments(
```

Comment:
I think we can't put these expression parser code in a separated class, just left TrinoMetadata clean.

We can make a new class parse from expression to paimon Predicate

Reply:
And we can add some UT test for this new class if possible.